precompiles/erc20/approve.go

```diff
@@ -211,26 +217,49 @@ func (p Precompile) createAuthorization(ctx sdk.Context, grantee, granter common
 	return p.AuthzKeeper.SaveGrant(ctx, grantee.Bytes(), granter.Bytes(), authorization, &expiration)
 }
 
-func (p Precompile) updateAuthorization(ctx sdk.Context, grantee, granter common.Address, amount *big.Int, authorization *banktypes.SendAuthorization) error {
-	authorization.SpendLimit = sdk.Coins{{Denom: p.tokenPair.Denom, Amount: sdk.NewIntFromBigInt(amount)}}
+func (p Precompile) updateAuthorization(ctx sdk.Context, grantee, granter common.Address, amount *big.Int, authorization *banktypes.SendAuthorization, expiration *time.Time) error {
+	found, denomAmount := authorization.SpendLimit.Find(p.tokenPair.Denom)
+	if found {
+		// NOTE: we are first removing the existing amount and then adding the new one
```

Comment:
This is very hard to follow, please refactor

Why are we removing the amount and then adding the same denom instead of replacing the amount for that denom?

Reply:
This was the easiest way I found to replace the denom amount? Is there a more straight forward approach?

The alternative I saw was to create a new `Coins` struct and then iterate over all `Coin` and add them unless they are the denomination to remove, but that seemed less intuitive to me

Reply:
okay there was definitely a simpler approach 😅